packages/medusa/src/api/middlewares/require-customer-authentication.ts

```diff
@@ -0,0 +1,12 @@
+import { NextFunction, Request, RequestHandler, Response } from "express"
+import passport from "passport"
+
+export default (): RequestHandler => {
+  return (req: Request, res: Response, next: NextFunction): void => {
```

Comment:
thought(non-blocking): when this is called the `authenticateCustomer` middleware will already be applied. This means the JWT is parsed. We could just verify with a simple `if` that the customer has been parsed. Not an important change but just wanted to highlight that this might be duplicate work.

Reply:
You are right, I would just say that a user could use them independently and therefore I am not sure we should go with the idea that we know that internally it has already been parsed. Wdyt?